### PR TITLE
Migrate from godep to Go modules

### LINKED ISSUE #3541: Migrate from dep to go modules
Go modules are the future. They works outside of GOPATH, so build instructions are simpler.

### PR DESCRIPTION
**Pull request by awly**
Some vendored dependency versions were changed because of transient dependency requirements.
You can see an example and exact version bumps in https://github.com/gravitational/teleport/commit/59956d4170780e322a6bfdc0df19ec20c1c3f31f.

Added readme instructions and removed old godep files to avoid confusion.

There's one edge case that requires https://github.com/gravitational/teleport.e/pull/143

Fixes https://github.com/gravitational/teleport/issues/3541

### CODE REVIEW

**Comment by webvictim** [comment]
LGTM. I'll let others who are more familiar with Go modules/vendoring review though.

**Review by fspmarshall — APPROVED** [review]
I don't have much experience with go modules, but this worked flawlessly for me.  I'm excited about this change.

**Comment by klizhentas** [comment]
@webvictim hey Gus, why is Drone stuck on this PR?

**Comment by webvictim** [comment]
For some reason Drone was waiting for approval for the `test-docs` pipeline - I have no idea why, given that the `test` pipeline ran OK and didn't require any manual intervention (at least from me)

Usually pipeline approval is needed when someone modifies the `.drone.yml` file and doesn't sign it, but it doesn't look like this PR touched it. Weird glitch maybe.

**Review by webvictim — APPROVED** [review]

**Comment by awly** [comment]
Need @russjones or @klizhentas approval for good measure

File: README.md
Comment (russjones) [inline README.md:144]: Capitalization for this and all titles.

**Review by russjones — APPROVED** [review]

**Comment by awly** [comment]
retest this please

**Pull request merged**
